packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -1242,7 +1243,18 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       this.checkLVal(node.id, BIND_TS_TYPE, undefined, "typescript type alias");
 
       node.typeParameters = this.tsTryParseTypeParameters();
-      node.typeAnnotation = this.tsExpectThenParseType(tt.eq);
+      node.typeAnnotation = this.tsInType(() => {
+        this.expect(tt.eq);
+
+        if (this.isContextual("intrinsic")) {
```

Comment:
If a dot token is after a contextual `"intrinsic"` it is not considered as an `TSIntrinsicKeyword`:

https://github.com/microsoft/TypeScript/pull/40580/files#diff-ead24f0f0f59c0ea9c1c511052e8884bR6745

```ts
namespace intrinsic {
    export class Foo {}
}
type Foo = intrinsic.Foo;
```